Index: lib/vendor/symfony/lib/widget/sfWidgetFormSchemaFormatter.class.php
===================================================================
--- lib/vendor/symfony/lib/widget/sfWidgetFormSchemaFormatter.class.php	(révision 33616)
+++ lib/vendor/symfony/lib/widget/sfWidgetFormSchemaFormatter.class.php	(copie de travail)
@@ -88,6 +88,10 @@
       return self::$translationCallable->call($subject, $parameters, $catalogue);
     }
 
+    // LIBRINFO - 29/08/2011
+    if ( is_callable('__') )
+      $this->setTranslationCallable('__');
+    
     return call_user_func(self::$translationCallable, $subject, $parameters, $catalogue);
   }
 
